src/components/tools/drawing-tool/drawing-tool.tsx

```diff
@@ -1,40 +1,35 @@
-import React from "react";
-import { BaseComponent } from "../../base";
-import { ToolTileModelType } from "../../../models/tools/tool-tile";
+import classNames from "classnames";
+import React, { useEffect } from "react";
+import { IToolTileProps } from "../tool-tile";
 import { ToolbarView } from "./drawing-toolbar";
 import { DrawingLayerView } from "./drawing-layer";
-import { TOOLBAR_WIDTH, DrawingContentModelType } from "../../../models/tools/drawing/drawing-content";
+import { useToolbarToolApi } from "../hooks/use-toolbar-tool-api";
+import { DrawingContentModelType } from "../../../models/tools/drawing/drawing-content";
 
 import "./drawing-tool.sass";
 
-interface IProps {
-  model: ToolTileModelType;
-  readOnly: boolean;
-  scale?: number;
-  onSetCanAcceptDrop: (tileId?: string) => void;
-}
+type IProps = IToolTileProps;
 
-export default class DrawingToolComponent extends BaseComponent<IProps> {
+const DrawingToolComponent: React.FC<IProps> = (props) => {
+  const { documentContent, toolTile, model, readOnly, onRegisterToolApi, onUnregisterToolApi } = props;
 
-  public static tileHandlesSelection = true;
-
-  public componentDidMount() {
-    if (!this.props.readOnly) {
-      (this.props.model.content as DrawingContentModelType).reset();
+  useEffect(() => {
+    if (!readOnly) {
+      (model.content as DrawingContentModelType).reset();
     }
-  }
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Is this another case where we get extra renders with the full set of dependencies?

Reply:
Although now I see there are several of these throughout the PR.  Is there any argument for removing the lint rule?

Reply:
I don't think so. Ignoring dependencies should be a conscious choice rather than a default. That said, there's a kind of "tyranny of exhaustive-deps" argument that suggests that one should never disable the rule and instead always figure out the right combination of `useCallback`/`useMemo`/`useRef` to achieve the desired result. I think the truth is somewhere in the middle -- it's important to think about whether you have the right dependencies, but if there are good reasons to believe that a given effect should only run once, disabling the rule makes more sense than jumping through hoops to achieve the same result. In the world of class components we routinely put things in `componentDidMount()`/`componentWillUnmount()` -- passing an empty dependencies array to `useEffect` is essentially the same thing. What I really want is for it to be possible to pass a second "ignore" array to `useEffect` which would disable the warning for dependencies individually but I'm not going to hold my breath.